app/assets/javascripts/discourse/app/lib/hashtag-types/category.js

```diff
@@ -12,29 +12,20 @@ export default class CategoryHashtagType extends HashtagTypeBase {
     return this.site.categories || [];
   }
 
-  generateColorCssClasses(category) {
-    const generatedCssClasses = [];
-    const backgroundGradient = [`var(--category-${category.id}-color) 50%`];
-    if (category.parentCategory) {
-      backgroundGradient.push(
-        `var(--category-${category.parentCategory.id}-color) 50%`
-      );
-    } else {
-      backgroundGradient.push(`var(--category-${category.id}-color) 50%`);
-    }
-
-    generatedCssClasses.push(`.hashtag-color--category-${category.id} {
-  background: linear-gradient(-90deg, ${backgroundGradient.join(", ")});
-}`);
-
-    return generatedCssClasses;
+  generateColorCssClasses() {
+    // We rely on default CSS styling
+    return [];
   }
 
   generateIconHTML(hashtag) {
-    const hashtagId = parseInt(hashtag.id, 10);
-    const colorCssClass = !this.preloadedData.mapBy("id").includes(hashtagId)
-      ? "hashtag-missing"
-      : `hashtag-color--${this.type}-${hashtag.id}`;
-    return `<span class="hashtag-category-badge ${colorCssClass}"></span>`;
+    if (hashtag.colors?.length) {
```

Comment:
I think we can do a bit better here.

Reply:
I think this is OK since there is not really a way we can get around sometimes using `hashtag.colors` (i.e. when we are not sure if the category is loaded yet) and sometimes using the badge category `data-category-id` style.

I will note though that this also needs to be done for the hashtag-types/channel.js as well